File: app/src/main/java/com/fight/covid/ui/Repository.java
package com.fight.covid.ui;

import com.fight.covid.model.Response;
import com.fight.covid.network.ApiCallInterface;

import io.reactivex.Observable;

public class Repository {
    private ApiCallInterface apiCallInterface;

    public Repository(ApiCallInterface apiCallInterface) {
        this.apiCallInterface = apiCallInterface;
    }

    /*
     * method to call login api
     * */
    public Observable<Response> loadData() {
        return apiCallInterface.loadData();
    }
}

File: app/src/main/java/com/fight/covid/ui/Repository.kt
package com.fight.covid.ui

import androidx.lifecycle.LiveData
import com.fight.covid.FightCovidApplication
import com.fight.covid.model.Response
import com.fight.covid.network.ApiCallInterface
import com.fight.covid.room.Countries
import com.fight.covid.room.CountriesDao
import com.fight.covid.room.CountriesRoomDatabase
import io.reactivex.Observable
import javax.inject.Inject


class Repository @Inject constructor( var apiCallInterface: ApiCallInterface?,
var countryDao: CountriesDao?) {


    /*
     * method to call login api
     * */
    fun loadData(): Observable<Response> {
        return apiCallInterface!!.loadData()
    }
    suspend fun insert(country: Countries){
        countryDao?.insert(country)
    }

   suspend fun getCountryByCode(countryCode:String):Countries?{
        return countryDao?.getCountryByCode(countryCode)
    }


}